centrallix-sysdoc/OSDriver_Authoring.md

```diff
@@ -115,497 +170,1032 @@ License:  Copyright (C) 2001-2011 LightSys Technology Services. See LICENSE.txt
 
-Normally, objectsystem drivers will be able to manage any number of 'node' objects and the subtrees rooted at them.  Each 'node' object will normally relate to a particular instance of a network resource, or in some cases, a group of resources that are easily enumerated.  For example, a POP3 server would be a network resource that an OS driver could be written for.  If the network had multiple POP3 servers, then that one OS driver would be able to access each of them using different node objects.  However, if somehow the OS driver were able to easily enumerate the various POP3 servers on the network (i.e., they responded to some kind of hypothetical broadcast query), then the OS driver author could optionally design the driver to list the POP3 servers under a single node for the whole network.
+Once a file is openned, the driver should organize provided data into a tree-structured hierarchy, which becomes part of the path used by Centrallix's ObjectSystem.  For example, when opening `example.qy` in the ObjectSystem, the driver makes `/rows` and `/columns` available, allowing for paths such as `/apps/data/example.qy/rows`.  The root of a driver's tree (`example.qy`) is called the driver's "node" object, and most paths traverse the root nodes of multiple drivers.  A driver author is free to define any manner of tree structures for representing data available within their driver.  However, the structure should fit the basic ObjectSystem model of a hierarchy of objects, each having attributes, and optionally some methods and/or content.
 
-The structure of the subtree beneath the node object is entirely up to the drivers' author to determine; the OSML does not impose any structural restrictions on such subtrees.
+A driver can be openned multiple times, leading one driver to have multiple "node" objects, also called instances.  Typically, each "node" object relates to a particular instance of a network resource.  For example, an instance of a POP3 driver might represent a POP3 server on the network.  If the network had multiple POP3 servers, this driver could be used to access each of them through different node objects (e.g. `dev.pop3`, `prod.pop3`, etc.).  However, if somehow the OS driver were able to easily enumerate the various POP3 servers on the network (i.e., they responded to some kind of hypothetical broadcast query), then the OS driver author could also design the driver to list the POP3 servers under a single node for the whole network.
```

Comment:
I'm not sure pop3 servers would be a very good example of something to automatically enumerate or to design a driver to auto-enumerate.  A better example might be IOT devices like in a home automation system or printers on a local network.  However, for security reasons, it can often be better to set up individual node objects rather than enumeration, so that access can be better limited.

Reply:
I copied the content of this example from the previous paragraph (although my restructuring for clarity means git doesn't show that very well).

Are you suggesting that we replace it with a different example?

Reply:
A different example using a database has been added. I think it's pretty good and I'll look forward to hearing your thoughts.